unixfs/io/bufdagreader.go

```diff
@@ -10,7 +10,9 @@ type bufDagReader struct {
 	*bytes.Reader
 }
 
-func NewBufDagReader(b []byte) *bufDagReader {
+// newBufDagReader returns a DAG reader for the given byte slice.
+// BufDagReader is used to read RawNodes.
+func newBufDagReader(b []byte) *bufDagReader {
```

Comment:
I'd just make `bufDatReader` public.

Reply:
So between increasing the API surface and decreasing it, I went with the latter (since this method is not used by anyone). But I'll make it public then...

Reply:
Yeah, but you exposed the `PBDagReader` so I figured we should be consistent at least.